chatsky/llm/filters.py

```diff
@@ -0,0 +1,93 @@
+"""
+Filters
+---------
+This module contains a collection of basic functions for history filtering to avoid cluttering LLMs context window.
+"""
+
+import abc
+from enum import Enum
+from pydantic import BaseModel
+
+from logging import Logger
+
+from chatsky.core.message import Message
+from chatsky.core.context import Context
+
+logger = Logger(name=__name__)
+
+
+class BaseHistoryFilter(BaseModel, abc.ABC):
+    """
+    Base class for all message history filters.
+    """
+
+    class Return(Enum):
+        Request = 1
+        Response = 2
+        Turn = 3
+
+    def call(self, ctx: Context, request: Message, response: Message, model_name: str):
+        return self.Return.Turn
+
+    def __call__(self, ctx: Context, request: Message, response: Message, model_name: str) -> bool:
+        """
+        :param ctx: Context object.
+        :param request: Request message.
+        :param response: Response message.
+        :param model_name: Name of the model in the Pipeline.models.
+        """
+        try:
+            result = self.call(ctx, request, response, model_name)
+        except Exception as exc:
+            logger.warning(exc)
+            return []
+        if result == self.Return.Turn:
+            return [request, response]
+        if result == self.Return.Response:
+            return [response]
+        if result == self.Return.Request:
+            return [request]
+        return []
+
+
+class MessageFilter(BaseHistoryFilter):
+    @abc.abstractmethod
+    def call(self, ctx, message, model_name):
+        raise NotImplemented
+
+    def __call__(self, ctx, request, response, model_name):
+        return self.call(ctx, request, model_name) + self.call(ctx, response, model_name)
```

Comment:
Needs to be bitwise or:
```suggestion
        return self.call(ctx, request, model_name) | self.call(ctx, response, model_name)
```
Add tests. They did not catch this.

Reply:
https://github.com/deeppavlov/chatsky/pull/376/commits/1ff7020da5257a0f1f14410b6a73df915911d2ac

Reply:
I can't find tests for that.